spec/support/object_space_helper.rb

```diff
@@ -0,0 +1,19 @@
+require 'objspace'
+
+module ObjectSpaceHelper
+  module_function
+
+  def estimate_bytesize_supported?
```

Comment:
Minor: I suggest not using a `module_function` for these kinds of helpers, and instead defining the functions on `self`. This way they always get prefixed with the module `ObjectSpaceHelper.estimate_bytesize_supported?`, rather than being included somewhere which is always quite confusing ("where is this method coming from?").